src/providers/twitch/TwitchIrcServer.cpp

```diff
@@ -935,31 +940,55 @@ void TwitchIrcServer::onReadConnected(IrcConnection *connection)
 {
     (void)connection;
 
-    std::lock_guard lock(this->channelMutex);
+    std::vector<ChannelPtr> activeChannels;
+    {
+        std::lock_guard lock(this->channelMutex);
 
-    // join channels
-    for (auto &&weak : this->channels)
+        activeChannels.reserve(this->channels.size());
+        for (const auto &weak : this->channels)
+        {
+            if (auto channel = weak.lock())
+            {
+                activeChannels.push_back(channel);
+            }
+        }
+    }
+
+    // get the selected channels
+    std::set<Channel *> selected;
+    for (auto *window : getApp()->getWindows()->allWindows())
     {
-        if (auto channel = weak.lock())
+        auto *page = window->getNotebook().getSelectedPage();
+        if (!page)
+        {
+            continue;
+        }
+
+        for (auto *split : page->getSplits())
         {
-            this->joinBucket_->send(channel->getName());
+            selected.emplace(split->getChannel().get());
         }
     }
 
+    // put the selected channels first
+    std::ranges::partition(activeChannels, [&](const auto &chan) {
```

Comment:
Could you swap to [std::ranges::stable_partition](https://en.cppreference.com/w/cpp/algorithm/ranges/stable_partition) to ensure we retain the relative order of the two segments? imo it's nice if you can somewhat know what order of the join of your inactive channels